### ISSUE TITLE
Make `xsv search` more like csvgrep

### ISSUE THREAD

**Issue by d33tah** [BurntSushi/xsv#90]
I recently wanted to use `xsv search` as a csvgrep substiute - the way I understood it, this is the subcommand closest to csvgrep. Here are three things I missed though:

1. `-f` / `--file`: load a list of patterns from 
2. `-F`, `--fixed-strings`: don't assume regex, consider input a fixed string,
3.  `-x`, `--line-regexp`: imply ^$ around regex - `test` becomes `^test$`

@BurntSushi any chance you could implement those or point me to a subcommand that already does it if I missed it? My workaround was something along the lines of ```"`</tmp/patterns.txt sed -e 's/^/^/g' -e 's/$/$/g'  | tr '\n' '|'  | head -c -1 `"```, but that doesn't look elegant.

**Comment by BurntSushi** [comment]
That functionality indeed does not exist. There isn't any particular reason, it just hasn't been done yet. I personally won't be working on it any time soon, but anyone is welcome to take a crack at it here: https://github.com/BurntSushi/xsv/blob/master/src/cmd/search.rs --- I'd be happy to answer questions.

Also, I'm not sure the name `--line-regexp` is correct. You might want to name that `--field-regexp` instead.

**Comment by d33tah** [comment]
You're right! I might play with it, but I can't promise anything since I'm super new to Rust. Thanks for the response!

**Comment by BurntSushi** [comment]
@d33tah Thanks! I'm happy to help in any way. Any question is OK, even if they are just general things about Rust. Feel free to use this issue, email me or ping me on IRC. :-)

**Comment by ghuls** [comment]
Reading a list of patterns (or in most cases plain strings) from a file would be very useful for me too.
A lot of bioinformatic fileformats are TAB-separated files which have e.g. gene names in a certain field, which you want to get out.

I wrote a bash/awk script that allows to search multiple patterns from a file in a certain column for this purpose:
https://github.com/ghuls/multigrep/blob/master/multigrep.sh

**Comment by unhammer** [comment]
In the meanwhile, `xsv search "$(tr '\n' '|' < pattern-lines.txt | sed 's/|$//')` does work in bash.
